src/test/java/org/folio/rest/impl/EHoldingsRootProxyImplTest.java

```diff
@@ -126,11 +96,97 @@ public void shouldReturnUnauthorizedWhenRMAPIRequestCompletesWith403ErrorStatus(
       .statusCode(403);
   }
   
-  private RequestSpecification getRequestSpecification() {
-    return TestUtil.getRequestSpecificationBuilder(host + ":" + port)
-      .addHeader(RestConstants.OKAPI_URL_HEADER,   host + ":" + userMockServer.port())
-    .build();
+  @Test
+  public void shouldReturnUpdatedProxyOnSuccessfulPut() throws IOException, URISyntaxException {
+    String stubResponseFile = "responses/rmapi/proxiescustomlabels/get-root-proxy-custom-labels-updated-response.json";
+
+    String wiremockUrl = getWiremockUrl();
+    TestUtil.mockConfiguration(CONFIGURATION_STUB_FILE, wiremockUrl);
+
+    WireMock.stubFor(
+      WireMock.get(new UrlPathPattern(new RegexPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/"), true))
+        .willReturn(new ResponseDefinitionBuilder().withBody(TestUtil.readFile(stubResponseFile))));
+
+    WireMock.stubFor(
+      WireMock.put(new UrlPathPattern(new RegexPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/"), true))
+        .willReturn(new ResponseDefinitionBuilder().withStatus(204)));
+
+    ObjectMapper mapper = new ObjectMapper();
+    RootProxyPutRequest rootProxyToBeUpdated = mapper.readValue(TestUtil.getFile("requests/kb-ebsco/put-root-proxy.json"),
+        RootProxyPutRequest.class);
+
+    RootProxy expected = getUpdatedRootProxy();
+    RequestSpecification requestSpecification = getRequestSpecification();
+
+    String updateRootProxyEndpoint = "eholdings/root-proxy";
+
+    RootProxy rootProxy = RestAssured
+      .given()
+      .spec(requestSpecification)
+      .header(CONTENT_TYPE_HEADER)
+      .body(mapper.writeValueAsString(rootProxyToBeUpdated))
```

Comment:
One minor thing, there is TestUtil#readFile that reads classpath file to string, so this will work too:
.body(TestUtil.readFile("requests/kb-ebsco/put-root-proxy.json"))